auslib/web/admin/views/rules.py

```diff
@@ -285,45 +283,60 @@ def get(self, column):
+        what = {}
+        for field in connexion.request.json:
+            if change_type == "insert" and field in ["rule_id", "data_version"]:
+                continue
 
-            self.log.debug("releaseNames: %s" % releaseNames)
-            self.log.debug("transaction: %s" % transaction)
+            if field in ["rule_id", "data_version"]:
+                what[field] = int(connexion.request.json[field])
+            else:
+                what[field] = connexion.request.json[field]
 
-            form.mapping.choices = [(item['name'], item['name']) for item in releaseNames]
-            form.mapping.choices.insert(0, ('', 'NULL'))
+        # Explicit checks for each change_type
+        if change_type in ["update", "delete"]:
+            for field in ["rule_id", "data_version"]:
+                if not what.get(field, None):
+                    return problem(400, "Bad Request", "Missing field", ext={"exception": "%s is missing" % field})
 
         elif change_type == "insert":
-            form = ScheduledChangeNewRuleForm()
+            for field in ["update_type", "backgroundRate", "priority"]:
+                if not what.get(field, None):
+                    return problem(400, "Bad Request", "Missing field", ext={"exception": "%s is missing" % field})
+        else:
+            return problem(400, "Bad Request", "Invalid or missing change_type")
 
-            releaseNames = dbo.releases.getReleaseNames(transaction=transaction)
+        if change_type in ["update", "insert"]:
+            rule_dict, mapping_values, fallback_mapping_values = process_rule_form(what)
+            what = rule_dict
+            if what.get('mapping') is not None and len(mapping_values) != 1:
+                return problem(400, 'Bad Request', 'Invalid mapping value. No release name found in DB')
 
-            self.log.debug("releaseNames: %s" % releaseNames)
-            self.log.debug("transaction: %s" % transaction)
-
-            form.mapping.choices = [(item['name'], item['name']) for item in releaseNames]
-            form.mapping.choices.insert(0, ('', 'NULL'))
+            if what.get('fallbackMapping') is not None and len(fallback_mapping_values) != 1:
+                return problem(400, 'Bad Request', 'Invalid fallbackMapping value. No release name found in DB')
 
+        # TODO: currently UI passes extra rule model fields in change_type == 'delete' request body. Fix it and
+        # TODO: change the below operation from filter/pop to throw Error when extra fields are passed.
         elif change_type == "delete":
-            form = ScheduledChangeDeleteRuleForm()
-
-        else:
-            return Response(status=400, response="Invalid or missing change_type")
+            for key in list(what):
+                if key not in ["telemetry_product", "telemetry_channel", "telemetry_uptake", "when", "rule_id",
+                               "data_version", "change_type"]:
+                    what.pop(key, None)
```

Comment:
Oh mea culpa. Rather than popping it out, simply ignoring the rest of fields while constructing 'what 'if change_type == 'delete'